sigstore/verify/verifier.py

```diff
@@ -635,3 +607,29 @@ def validate_hashedrekord_v002_entry_body(bundle: Bundle) -> None:
         raise VerificationError(
             "transparency log entry is inconsistent with other materials"
         )
+
+
+def _v2_verifier_from_certificate(certificate: Certificate) -> v2.Verifier:
+    public_key = certificate.public_key()
+    key_details = None
+
+    if isinstance(public_key, EllipticCurvePublicKey):
+        if public_key.curve.name == "secp256r1":
+            key_details = cast(
+                v1.PublicKeyDetails,
+                v1.PublicKeyDetails.PKIX_ECDSA_P256_SHA_256,
+            )
+        else:
+            raise ValueError(f"Unsupported EC curve: {public_key.curve.name}")
+
+    # TODO support other keys
```

Comment:
Do we have blessed list of key types we should expect the signing certificates to contain?

Reply:
https://github.com/sigstore/architecture-docs/blob/main/algorithm-registry.md is close. I will have a closer look.